### PR TITLE
[CFM-475] notify group admins that status has changed to unpublished.

### PR DESCRIPTION
**Pull request by anvmn**
## This PR is dependent on #1059 

https://issuetracker.amplexor.com/jira/browse/CFM-475

### CODE REVIEW

**Comment by anvmn** [comment]
selection_113

**Comment by ordavidil** [comment]
@anvmn 
please re-roll

**Comment by anvmn** [comment]
@ordavidil 
re-rolled

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc [hunk 1]
```diff
@@ -74,9 +74,33 @@ function c4m_user_notifications_group_status_set_republished($node, $prev_status
...
+  }
+}
+
+/**
+ * Helper function.
+ *
+ * Email notification to organic group admin / owner, about status change
+ * from 'published' to 'draft'.
+ */
+function _c4m_user_notifications_to_admin_group_status_set_unpublished($node, $admin_id, $prev_status) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc]: => `$previous_status`

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc [hunk 2]
```diff
@@ -74,9 +74,33 @@ function c4m_user_notifications_group_status_set_republished($node, $prev_status
...
+  $message = message_create('c4m_group_unpublished', array('uid' => $admin_id));
+  c4m_message_load_tokens__c4m_group_unpublished($message);
+  $message->arguments['@group_former_status'] = $prev_status;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc]: @anvmn 
let's be consistent...  use `@group_previous_status` instead ...

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc [hunk 3]
```diff
@@ -74,9 +74,33 @@ function c4m_user_notifications_group_status_set_republished($node, $prev_status
...
- * Email notification about organic group status change to 'unpublished'.
+ * Email notification about organic group that was 'unpublished'.
+ *
+ * Status changed from 'published' to 'draft'.
  */
 function c4m_user_notifications_group_status_set_unpublished($node, $prev_status) {
+  $group_admins_ids = _c4m_user_notifications_get_admins_and_owner_ids($node->nid);
+
+  foreach ($group_admins_ids as $admin_id) {
+    _c4m_user_notifications_to_admin_group_status_set_unpublished($node, $admin_id, $prev_status);
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc]: => `$group_former_status`

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc [hunk 4]
```diff
@@ -74,9 +74,33 @@ function c4m_user_notifications_group_status_set_republished($node, $prev_status
 }
 
 /**
- * Email notification about organic group status change to 'unpublished'.
+ * Email notification about organic group that was 'unpublished'.
+ *
+ * Status changed from 'published' to 'draft'.
  */
 function c4m_user_notifications_group_status_set_unpublished($node, $prev_status) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc]: => `$group_former_status`

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc [hunk 5]
```diff
@@ -74,9 +74,33 @@ function c4m_user_notifications_group_status_set_republished($node, $group_forme
...
+ */
+function _c4m_user_notifications_to_admin_group_status_set_unpublished($node, $admin_id, $group_former_status) {
+  $message = message_create('c4m_group_unpublished', array('uid' => $admin_id));
+  c4m_message_load_tokens__c4m_group_unpublished($message);
+  $message->arguments['@group_former_status'] = $group_former_status;
+
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $wrapper->field_group_node->set($node->nid);
+  // Message will be saved in message_notify_send_message().
+  message_notify_send_message($wrapper->value());
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc]: `message_subscribe_send_message()`

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 6]
```diff
@@ -78,6 +79,20 @@ function _c4m_message_format_welcome_to_group_from_field(&$message, $message_ent
 /**
  * Helper function.
  *
+ * Formats 'c4m_group_unpublished' message From field, per template requirement.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: @anvmn 
Can you please write the ID of the template ?
thanks

**Comment by anvmn** [comment]
## Replaced by #1189

**Pull request closed without merging**